sha2/sha2.c

```diff
@@ -470,7 +471,7 @@ void dtls_sha256_transform(dtls_sha256_ctx* context, const sha2_word32* data) {
 
 #else /* SHA2_UNROLL_TRANSFORM */
 
-void dtls_sha256_transform(dtls_sha256_ctx* context, const sha2_word32* data) {
+void dtls_sha256_transform(dtls_sha256_ctx* context, const sha2_byte* data) {
```

Comment:
The change from `sha2_word32` to `sha2_byte` (i.e. from `uint32_t` to `uint8_t`) breaks the `data++` if `BYTE_ORDER != LITTLE_ENDIAN`.

Reply:
Missed that in my testing as I was not running with BIG_ENDIAN code.  However, I suspect that the same issue (non aligned word access giving unexpected results) will be present on BIG ENDIAN code as well.  Looks like a lot of code re-writing needed.

[Finding the issue was an interesting time!]

Reply:
Indeed.

The big endian code at least has not the issue with the REVERSE macros as they are no-ops except for the pointer increment. IMO your fix for `dtls_sha256_transform()` is a good start and just needs a small adjustment for the big endian case. (Also note that the type cast of `context->buffer` at the very beginning is a potential pitfall as it works on the assumption that the buffer always starts at a word boundary.)

**But** there are more alignment issues in the unrolled version of `dtls_sha265_transform()`. Uh!

Reply:
I have not fixed the W256 and W512 typecast for `context->buffer` as I think it very unlikely that they would not be the appropriate word aligned and changing the code would be complex to convert to byte aligned.